manifest/v1alpha/slo/validation_composite_test.go

```diff
@@ -0,0 +1,249 @@
+package slo
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/nobl9/nobl9-go/internal/testutils"
+	"github.com/nobl9/nobl9-go/internal/validation"
+	"github.com/nobl9/nobl9-go/manifest"
+	"github.com/nobl9/nobl9-go/manifest/v1alpha"
+)
+
+func TestValidate_CompositeSLO(t *testing.T) {
+	t.Run("passes", func(t *testing.T) {
+		slo := validCompositeSLO()
+		err := validate(slo)
+		testutils.AssertNoError(t, slo, err)
+	})
+	t.Run("fails - spec.indicator provided", func(t *testing.T) {
+		for _, ind := range []Indicator{
+			{
+				MetricSource: MetricSourceSpec{Name: "name-only"},
+			},
+			{
+				MetricSource: MetricSourceSpec{
+					Name:    "name",
+					Project: "default",
+					Kind:    manifest.KindAgent,
+				},
+			},
+			{
+				MetricSource: MetricSourceSpec{
+					Name:    "name",
+					Project: "default",
+					Kind:    manifest.KindDirect,
+				},
+			},
+		} {
+			slo := validCompositeSLO()
+			slo.Spec.Indicator = ind
+			err := validate(slo)
+			testutils.AssertContainsErrors(t, slo, err, 1, testutils.ExpectedError{
+				Prop:    "spec.indicator",
+				Code:    validation.ErrorCodeForbidden,
+				Message: "property is forbidden; indicator section is forbidden when spec.objectives[0].composite is provided",
+			})
+		}
+	})
+	t.Run("fails - other objective types mixed with composite", func(t *testing.T) {
+		for _, obj := range []Objective{
```

Comment:
Loop is used for a single element. We could add another one to test `CountMetricsSpec`.